File: programming 1/lab 5/src/commands/Command.java
package commands;

/**
 * Абстрактный класс, представляющий команду.
 */
public abstract class Command {

    /**
     * Выполняет команду с переданным аргументом.
     * @param arg аргумент команды.
     */
    abstract public void execute(String arg);
    private final String name;
    private final String descr;
    Command(String name, String descr) {
        this.name = name;
        this.descr = descr;
    }

    public String getName() {
        return name;
    }
    public String getDescr() {
        return descr;
    }
}

File: programming 1/lab 6 kotlin/kotlin/lab 6/src/main/java/server/commands/Command.kt
package server.commands

import common.communication.Response

/**
 * Абстрактный класс, представляющий команду.
 */
abstract class Command internal constructor(val name: String, val descr: String) {
    /**
     * Выполняет команду с переданным аргументом.
     * @param arg аргумент команды.
     */
    abstract fun execute(arg: String?): Response
}
